### ISSUE TITLE
pagebrowser shows on every page all records of repository

### ISSUE THREAD

**Issue by lisardo2** [jweiland-net/kk_downloader#2]
Hello, 

as you can see here: 
https://2019.feuerwerk-vpi.de/presse/mediathek/
every page shows the same set of records (all 26). The page browser works (but only with deactivated chash in install tool) but the records are not selected. 

After a quick look in the code i presume that it results from a discrepancy between the old  the old pibased page browser which is used in the fluid templates and the extbase repository classes. The repository class always selects all records and not a subset for every page. This is correct for extbase but only when the extbase pagebrowser is used who is able to lazy load the correct number of records.

**Comment by lisardo2** [comment]
Sorry: TYPO3 8 LTS and version 4.0.0.

**Comment by froemken** [comment]
Hello @lisardo2 

can you please try this PR?
https://github.com/jweiland-net/kk_downloader/pull/3

I have added limit and offset there. If it works for you I will release a new version

Have a nice start into this week

Stefan

**Comment by Schweriner** [comment]
I've adapted your PR but added a count all which is required
https://github.com/jweiland-net/kk_downloader/pull/11

**Comment by froemken** [comment]
The previous PageBrowser was a catastrophic. In earlier versions kk_downloader fetches ALL records by default, regardless, if only 2 or 20 records should be shown. For now I have added limit and offset to getDownloads, but I also had to change all `count($downloads)` parts. I have created a new method in DownloadRepository for that. Maybe I rework all this *** while removing TYPO3 9 compatibility ;-)
